hw/ip/kmac/data/kmac_base_testplan.hjson

```diff
@@ -5,27 +5,129 @@
+              - Signal cmd.done to tell KMAC to clear internal state.
+              - Try reading output digest again, confirm that it is 0.
             '''
       milestone: V1
       tests: ["{name}_smoke"]
     }
     {
-      name: feature1
-      desc: '''Add more test entries here like above.'''
-      milestone: V1
-      tests: []
+      name: long_message_and_output
+      desc: '''
+            Same as the smoke test, except with a message of up to 20KB.
+            Allow output length to vary up to 1KB (for XOF functions).
+            If output length is greater than 1600 bits, keccak rounds will be run manually to
+            squeeze extra output data.
+            Enable full 32-bit N, 256-bit S.
+            '''
+      milestone: V2
+      tests: ["{name}_long_message_and_output"]
+    }
+    {
+      name: burst_wr
+      desc: '''
+            This is the same as the long_message test, except we burst-write chunks of the message
+            into the msg_fifo, and disable intermediate status/CSR checks.
+            '''
+      milestone: V2
+      tests: ["{name}_burst_wr"]
+    }
+    {
+      name: test_vectors
+      desc: '''
+            These tests drive NIST test vectors for SHA3/SHAKE/CSHAKE/KMAC into the design and check
+            the output against the expected digest values.
+            '''
+      milestone: V2
+      tests: ["{name}_sha3_test_vectors", "{name}_shake_test_vectors", "{name}_cshake_test_vectors",
+      "{name}_test_vectors", "{name}_xof_test_vectors"]
+    }
+    {
+      name: sideload
+      desc: '''
+            Same as long_message_and_output test, except we set cfg.sideload and provide a
+            valid sideloaded key as well as a valid SW-provided key.
+            KMAC should operate on the sideloaded key.
+            '''
+      milestone: V2
+      tests: ["{name}_sideload"]
+    }
+    {
+      name: kdf
+      desc: '''
+            Test KeyMgr can initiate the KMAC operation through the KDF interface.
+            Use the keymgr_kmac_agent to send message data to the KMAC and to control the hashing
+            logic, and set cfg.sideload.
+            The result digest sent to the keymgr_kmac_agent will be compared against the result from
+            the C++ reference model.
+            In addition, try to read from the STATE window, confirm that this access is blocked
+            (return 0).
+            '''
+      milestone: V2
+      tests: ["{name}_kdf"]
+    }
+    {
+      name: kdf_with_sw_access
+      desc: '''
+            Initiate a KDF operation as before, but this time also provide a "SW" message and key,
+            and try to also initiate the KMAC with the CMD register.
+            KMAC should ignore the "SW" commands/data and proceed with KDF.
+            After KDF is done, trigger a software-initiated KMAC process using the previously
+            available message/key.
```

Comment:
This also could be added to the error code. I will note that and add it later into the design. Thanks